app/src/editor/canvas/components/DraggableCanvasWindow.jsx

```diff
@@ -35,74 +36,127 @@ export const Toolbar = ({ children, className }) => (
+
+            return {
+                ...layout,
+                clientX: layout.clientX + coreEvent.deltaX,
+                clientY: layout.clientY + coreEvent.deltaY,
+            }
         })
-    }
-
-    onDrag = (e, coreEvent) => {
-        if (!this.state.dragging) { return false }
-
-        this.setState({
-            clientX: this.state.clientX + coreEvent.deltaX,
-            clientY: this.state.clientY + coreEvent.deltaY,
-        })
-    }
+    }, [setLayoutState])
 
-    onDragStop = () => {
-        if (!this.state.dragging) { return false }
+    const onPositionUpdateProp = props.onPositionUpdate
+    const onDragStop = useCallback(() => {
+        if (!layout.dragging) { return }
 
-        this.setState({
+        setLayoutState({
+            ...layout,
             dragging: false,
         })
 
-        if (this.props.onPositionUpdate) {
-            this.props.onPositionUpdate(this.state.clientX, this.state.clientY)
+        if (onPositionUpdateProp) {
+            onPositionUpdateProp(layout.clientX, layout.clientY)
         }
-    }
+    }, [layout, setLayoutState, onPositionUpdateProp])
+
+    const style = useMemo(() => ({
+        left: layout.clientX,
+        top: layout.clientY,
+        width: layout.width,
+        height: layout.height,
+    }), [layout])
+
+    const { width, height } = layout
+
+    const onResizeStart = useCallback(() => {
+        setLayoutState((layout) => ({
+            ...layout,
+            resizing: true,
+        }))
+    }, [setLayoutState])
+
+    const onResize = useCallback((e, data) => {
+        setLayoutState((layout) => ({
+            ...layout,
+            height: data.size.height,
+            width: data.size.width,
+        }))
+    }, [setLayoutState])
+
+    const onSizeUpdateProp = props.onSizeUpdate
+    const onResizeStop = useCallback(() => {
+        if (!layout.resizing) { return }
+
+        setLayoutState({
+            ...layout,
+            resizing: false,
+        })
 
-    render() {
-        const style = {
-            left: this.state.clientX,
-            top: this.state.clientY,
+        if (onSizeUpdateProp) {
+            onSizeUpdateProp(layout.width, layout.height)
         }
-
-        return (
-            <CanvasWindow>
-                <DraggableCore
-                    handle={`.${styles.titleContainer}`}
-                    onStart={this.onDragStart}
-                    onDrag={this.onDrag}
-                    onStop={this.onDragStop}
+    }, [layout, setLayoutState, onSizeUpdateProp])
+
+    return (
+        <CanvasWindow>
+            <DraggableCore
+                handle={`.${styles.titleContainer}`}
+                onStart={onDragStart}
+                onDrag={onDrag}
+                onStop={onDragStop}
+            >
+                <Resizable
+                    className={styles.ResizableBox}
+                    width={width}
+                    height={height}
+                    minConstraints={[200, 100]}
```

Comment:
It's possible that the value for the height should be a little higher. Otherwise the buttons may end out outside of the box.

<img width="267" alt="Screenshot 2019-06-03 09 32 58" src="https://user-images.githubusercontent.com/320066/58784917-bfe4ee80-85e4-11e9-8175-381c9b790846.png">

Reply:
Good catch, added 200px as min height